src/node.cpp

```diff
@@ -202,7 +202,7 @@ Node::~Node()
     // sync: remove reference from local filesystem node
     if (localnode)
     {
-        localnode->deleted = true;
+        localnode->reactToNodeChange(true);
```

Comment:
Is it safe to call this function from the end of the Node destructor?  If the localnode tries to access any info in the Node, it may well be referring to deleted (but non-NULL) pointers.

Reply:
I think it's fine. `reactToNodeChange` only access data in localnode and sync.

Reply:
can we set localnode->node = NULL first?  So that anyone who tries that in future will find out quickly.  It's the next line anyway

Reply:
Sure